### ISSUE TITLE
Multiple source of PHPMD version identification may lead to confusion

### ISSUE THREAD

**Issue by llaville** [phpmd/phpmd#1285]
- PHPMD version: 3.x-dev 3514060
- PHP Version: 8.2.30 (but not important in this case)
- Installation type: composer (2.9.x)
- Operating System / Distribution & Version: Windows 11 / WSL2 / Docker CE

### Current Behavior
CLI will pick version from CHANGELOG file (if available) => https://github.com/phpmd/phpmd/blob/3.x/src/TextUI/Command.php#L398, and fallback to `@package_version@` pattern string

But this pattern string is also defined by [`PHPMD::VERSION`](https://github.com/phpmd/phpmd/blob/3.x/src/PHPMD.php#L31) class constant too !
Used by Renderer(s)

If there are not synchronized version applied, there will be confusion to identify PHPMD

Actually to proove this concept, I've modified (in my fork) CHANGELOG as follow : 

```
diff --git a/CHANGELOG b/CHANGELOG
index 459c855b..48e9e47f 100644
--- a/CHANGELOG
+++ b/CHANGELOG
@@ -1,3 +1,5 @@
+phpmd-3.0.x
+
 phpmd-2.15.0 (2023/12/11)
 ========================
```

When I run `bin/phpmd --version`, I got expected result `PHPMD 3.0.x`

But when I use a renderer based on JSON (for example):  `bin/phpmd analyze /path/to/source --format=json`
I got such contents (head chunk) 

```json
{
    "version": "@package_version@",
    "package": "phpmd",

```

### Expected Behavior
Have only one unique source of version identification

## Checks before submitting
* [x] Be sure that there isn't already an issue about this. See: [Issues list](https://github.com/phpmd/phpmd/issues)
* [x] Be sure that there isn't already a pull request about this. See: [Pull requests](https://github.com/phpmd/phpmd/pulls)
* [ ] I have added every step to reproduce the bug.
* [ ] If possible I added relevant code examples.
* [ ] This issue is about 1 bug and nothing more.
* [ ] The issue has a descriptive title. For example:  "JSON rendering failed on Windows for filenames with space".

**Comment by llaville** [comment]
I found a side effect with custom renderer (feature implemented by my request https://github.com/phpmd/phpmd/issues/1196)

While I make intensive tests on current version PHPMD 3.0-dev (latest commit https://github.com/phpmd/phpmd/commit/35140602568b6068884b28e04b3dcf1cde9d5ea1), I found again a source of confusion.

As author of https://github.com/llaville/sarif-php-converters project, I run my custom Renderer 
- documentation : https://github.com/llaville/sarif-php-converters/blob/master/docs/converter/phpmd.md 
- source code: https://github.com/llaville/sarif-php-converters/blob/master/src/Converter/Reporter/PhpMdRenderer.php

And what a surprise when I detect that both following commands are not equivalent : 

Without `report-file` option : 

```shell
bin/phpmd analyze src/ --format='\Bartlett\Sarif\Converter\Reporter\PhpMdRenderer' --bootstrap=autoload.php > issue.sar.json
```

With `report-file` option : 

```shell
bin/phpmd analyze src/ --format='\Bartlett\Sarif\Converter\Reporter\PhpMdRenderer' --bootstrap=autoload.php --reportfile-sarif=issue.sarif.json
```

Both contents `issue.sar.json` and `issue.sarif.json` are not same !

Reason:  when you use a `report-file` option with a default format supported by PHPMD (i.e: `sarif`), it will generate the report corresponding to the related Renderer (here it's https://github.com/phpmd/phpmd/blob/3.x/src/Renderer/SARIFRenderer.php)
and NOT the version produced by the custom renderer.

Code that prooved this concept is located at :  
https://github.com/phpmd/phpmd/blob/3.x/src/TextUI/Command.php

```php
        // Create renderer and configure output
        $renderer = $options->createRenderer($output);
        $renderer->setWriter($output);
        $renderers = [$renderer];

        foreach ($options->getReportFiles() as $reportFormat => $reportFile) {
            $reportRenderer = $options->createRenderer($output, $reportFormat);
            $stream = fopen($reportFile, 'wb');
            if (!$stream) {
                throw new InvalidArgumentException("Unable to write to: '{$reportFile}'.");
            }
            $reportRenderer->setWriter(new StreamOutput($stream));

            $renderers[] = $reportRenderer;
        }
```

Results produced on stdOut and `--reportfile-sarif` are not the same

For example:  on stdout I got (head chunk) not prettified contents

```
{"$schema":"https://json.schemastore.org/sarif-2.1.0.json","version":"2.1.0","runs":[{"tool":{"driver":{"name":"PHPMD","shortDescription":{"text":"Tool with aims to be a PHP equivalent of the well known Java tool PMD"},"fullDescription":{"text":"PHPMD is a spin-off project of PHP Depend and aims to be a PHP equivalent of the well known Java tool PMD."},"version":"3.9999999.9999999.9999999-dev","semanticVersion":"3.9999999.9999999.9999999-dev","informationUri":"https://github.com/phpmd/phpmd","rules":[{"id":"CleanCodeRules/StaticAccess","name":"StaticAccess",

```
> [!NOTE]
> The `version` and `semanticVersion` are as expected

But on `issue.sarif.json` file created by the `report-file` option I got pretty prints contents

```json
{
    "version": "2.1.0",
    "$schema": "https:\/\/raw.githubusercontent.com\/oasis-tcs\/sarif-spec\/master\/Schemata\/sarif-schema-2.1.0.json",
    "runs": [
        {
            "tool": {
                "driver": {
                    "name": "PHPMD",
                    "informationUri": "https:\/\/phpmd.org",
                    "version": "@package_version@",
                    "rules": [

```

PS: Code that demonstrate this feature is on my branch `1.6` (especially with commit https://github.com/llaville/sarif-php-converters/commit/31956171b6d8de0f45f9477af965b5e17ca1f503)
